sdk/storage/Azure.Storage.DataMovement/tests/TestProgressHandler.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+
+namespace Azure.Storage.DataMovement.Tests
+{
+    internal class TestProgressHandler : IProgress<StorageTransferProgress>
```

Comment:
So, I ended up refactoring out the assertions into new static class but kept this very simple class for tracking the progress reports. I did try using a `Mock` and it added a bit of complications to the assertions (probably fine), but I also found that `Invocations` wasn't always in the order I was expecting (which is key for the assertions). I know there are ways to check order in `Moq` but decided it wasn't worth the extra complexity for now. The way it was refactored though makes it pretty easy to replace with a Mock if we want to go that route in the future.

I did also refactor how pause assertions work a bit to be able to handle multiple pauses since you called that out specifically :)